### PR TITLE
feat: add create_jwt() endpoint

### PR DESCRIPTION
**Pull request by ppetit-gitguardian**
This PR adds the SaaS-only `create_jwt()` endpoint.

### CODE REVIEW

**Comment by pierrelalanne** [comment]
Looks nice 👍

File: pygitguardian/client.py [hunk 1]
```diff
@@ -514,3 +516,29 @@ def read_metadata(self) -> Optional[Detail]:
...
+        """
+
+        resp = self.post(
+            endpoint="auth/jwt",
+            data={"audience": jwt_audience, "audience_type": service.value},
+            extra_headers=extra_headers,
+        )
+        obj: Union[Detail, JWTResponse]
+        if is_ok(resp):
+            obj = cast(JWTResponse, JWTResponse.SCHEMA.load(resp.json()))
```
Comment (agateau-gg) [inline pygitguardian/client.py:540]: I just merged some changes in py-gitguardian which you can take advantage of here: if you make `JWTResponse` inherit from `FromDictMixin` in addition to `Base`, then this code can be rewritten as:

```
obj = JWTResponse.from_dict(resp.json())
```

This is simpler and more type safe.

File: pygitguardian/client.py [hunk 2]
```diff
@@ -514,3 +516,29 @@ def read_metadata(self) -> Optional[Detail]:
...
+    def create_jwt(
+        self,
+        jwt_audience: str,
+        service: JWTService,
+        extra_headers: Optional[Dict[str, str]] = None,
+    ) -> Union[Detail, JWTResponse]:
+        """
+        Create a JWT token for other GitGuardian services.
+
+        Only used to interact with HasMySecretLeaked for now.
```
Comment (agateau-gg) [inline pygitguardian/client.py:529]: There is a risk this comment won't be removed when we add more services :). I think it's safer to add it to the changelog fragment.

**Review by agateau-gg — CHANGES_REQUESTED** [review]
Looks good 👍🏻. Just some minor remarks.

**Review by agateau-gg — APPROVED** [review]
Thanks, looks good to go now!

**Pull request merged**